test/join.jl

```diff
@@ -355,8 +355,8 @@ end
     @test levels(leftjoin(A, B, on=:b).b) == ["d", "c", "b", "a"]
     @test levels(leftjoin(B, A, on=:b).b) == ["a", "b", "c"]
     @test levels(rightjoin(A, B, on=:b).b) == ["a", "b", "c"]
-    @test levels(rightjoin(B, A, on=:b).b) == ["d", "c", "b", "a"]
```

Comment:
@nalimilan - these changes follow CategoricalArrays.jl 0.10.0 release, so a check would be welcome.

Reply:
Interesting. Why aren't levels the same for inner and outer joins? `["d", "c", "b", "a"]` would sound like a good choice since `levels(A.b)` is a superset of `levels(B.b)`.

That said, since the old behavior wasn't ideal either, the new one is OK -- both are kind of arbitrary.